### DISCUSSION TITLE
Is it possible to do a simple health check of RTSP stream with curl tool?

### DISCUSSION THREAD

**Discussion by helloworld20001000** [bluenviron/mediamtx#1605] [Questions]
Hi, I would like to check if the server RTSP is up or down by using command curl. If it isn't possible, is there any way to know its current status without needing to consume streaming with for example ffmpeg or vlc?.
I'm running the server RTSP in a docker container and when it is up if I execute:
$ curl --connect-timeout 15 -i -X OPTIONS rtsp://localhost:8554/mystream
it returns curl: (85) Empty reply from server
I checked this error and it indicates "Mismatch of RTSP CSeq numbers" and I don't know how to interpret it.
I the container with the server RTSP is down and I execute the same command, it returns:
curl: (7) Failed to connect to localhost port 8554: Connection refused

**Comment by tbnguyen1407** [comment]
I don't think there is a reliable way with only `curl`. If you want to use `ffprobe` you can reference this to get the stream info from SDP or show `unavailable` otherwise

```
$ FFPROBE_PARAMS="-v fatal -hide_banner -rtsp_transport tcp -stimeout 10000000 -of compact=print_section=0:item_sep=,:nokey=1 -show_entries stream=codec_name,width,height,r_frame_rate -select_streams v"
$ ffprobe $FFPROBE_PARAMS $STREAM_URL || echo unavailable
```

![amt-streams](https://user-images.githubusercontent.com/2490928/228598255-18aabe62-3d9a-4787-a777-b6ed49ca813a.png)

**Reply by helloworld20001000** [reply]
Thank u! I found a module in python (rtsp-curl) and used its function options() to know if a server rtsp is still up.

**Comment by aler9** [comment]
Hello, you can get health of all configured streams by enabling the API: 

```
docker run --rm -it \
--network=host \
--privileged \
--tmpfs /dev/shm:exec \
-v /run/udev:/run/udev:ro \
-e MTX_API=true \
aler9/rtsp-simple-server:latest
```

The API can be queried with CURL on port 9997:

```
curl http://localhost:9997/v1/paths/list
```

obtaining

```json
{
   "items":{
      "mystream":{
         "confName":"~^.*$",
         "conf":{
            ...
         },
         "source":{
            "type":"rtspSession",
            "id":"06b2e72f-81a3-437a-915c-dc13c9d49bef"
         },
         "sourceReady":true,
         "tracks":[
            "H264"
         ],
         "bytesReceived":49963,
         "readers":[
            {
               "type":"hlsMuxer"
            }
         ]
      }
   }
}
```

**Reply by helloworld20001000** [reply]
Thank u! That's another way to do it. I enabled the API to configure a proxy server but I didnt realize that I could have used that too to know if a server was still up. I found a module in python (rtsp-curl) and used its function options() to know if a server rtsp is still up.

**Comment by johnvoipguy** [comment]
curl http://localhost:9997/v1/rtspconns/list would be another option. it will show bytes sent/received.


    "items": {
        "38030c2e-8081-475b-86b9-c3e83b0c85a9": {
            "created": "2023-04-13T21:27:08.942921764-05:00",
            "remoteAddr": "127.0.0.1:43874",
            "bytesReceived": 3792,
            "bytesSent": 140897898
        },
        "4afa4540-1fba-425d-b9e8-1a50d73d7c05": {
            "created": "2023-04-13T21:27:04.982104597-05:00",
            "remoteAddr": "127.0.0.1:49294",
            "bytesReceived": 3765,
            "bytesSent": 200820835
        },

**Comment by alexandervlpl** [comment]
In my case MediaMTX is proxying a few cameras, and `curl -v -X DESCRIBE $URL` _seems_ to work, but doesn't! I actually get a 200 response even when a camera is offline. Definitely use one of the nice options here instead of `curl`.

[a comment by an automated account is not shown]
